src/test/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incidents/IncidentsNomisApiMockServer.kt

```diff
@@ -0,0 +1,149 @@
+          .withStatus(status),
+      ),
+    )
+  }
+
+  fun stubMultipleGetIncidentIdCounts(totalElements: Long, pageSize: Long) {
+    // for each page create a response for each incident id starting from 1 up to `totalElements`
+
+    val pages = (totalElements / pageSize) + 1
+    (0..pages).forEach { page ->
+      val startIncidentId = (page * pageSize) + 1
+      val endIncidentId = min((page * pageSize) + pageSize, totalElements)
+      nomisApi.stubFor(
+        get(
+          urlPathEqualTo("/incidents/ids"),
+        )
+          .withQueryParam("page", equalTo(page.toString()))
+          .willReturn(
+            aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+              .withBody(
+                incidentIdsPagedResponse(
+                  totalElements = totalElements,
+                  ids = (startIncidentId..endIncidentId).map { it },
+                  pageNumber = page,
+                  pageSize = pageSize,
+                ),
+              ),
+          ),
+      )
+    }
+  }
+
+  fun stubGetIncident(nomisIncidentId: Long = 1234) {
+    nomisApi.stubFor(
+      get(
+        urlPathEqualTo("/incidents/$nomisIncidentId"),
+      )
+        .willReturn(
+          aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+            .withBody(incidentResponse(nomisIncidentId)),
+        ),
+    )
+  }
+
+  fun stubGetIncidentNotFound(nomisIncidentId: Long = 1234) {
+    nomisApi.stubFor(
+      get(
+        urlPathEqualTo("/incidents/$nomisIncidentId"),
+      )
+        .willReturn(
+          aResponse().withHeader("Content-Type", "application/json")
+            .withStatus(HttpStatus.NOT_FOUND.value()),
+        ),
+    )
+  }
+
+  fun stubMultipleGetIncidents(intProgression: IntProgression) {
+    (intProgression).forEach {
+      nomisApi.stubFor(
+        get(
+          urlPathEqualTo("/incidents/$it"),
+        )
+          .willReturn(
+            aResponse().withHeader("Content-Type", "application/json").withStatus(HttpStatus.OK.value())
+              .withBody(incidentResponse(nomisIncidentId = it.toLong())),
+          ),
+      )
+    }
+  }
+
+  fun verify(pattern: RequestPatternBuilder) = nomisApi.verify(pattern)
+  fun verify(count: Int, pattern: RequestPatternBuilder) = nomisApi.verify(count, pattern)
+  fun verify(countMatchingStrategy: CountMatchingStrategy, requestPatternBuilder: RequestPatternBuilder) = nomisApi.verify(countMatchingStrategy, requestPatternBuilder)
+}
+
+fun incidentIdsPagedResponse(
+  totalElements: Long = 10,
+  ids: List<Long> = (0L..10L).toList(),
+  pageSize: Long = 10,
+  pageNumber: Long = 0,
+): String {
+  val content = ids.map { """{ "incidentId": $it }""" }.joinToString { it }
+  return pageContent(content, pageSize, pageNumber, totalElements, ids.size)
+}
+
+private fun incidentResponse(
```

Comment:
mentioned on previous PR I think prefer using DTOs generated from OpenAPI

Reply:
oo - think that was on csip - thought I'd captured them all here too - will update